app/models/data_export.rb

```diff
@@ -0,0 +1,79 @@
+require "csv"
+
+class DataExport
+  SUPPORTED_TYPES = %w(
+    Donation
+    DonationSite
+    Partner
+    Purchase
+    Distribution
+    DiaperDriveParticipant
+    StorageLocation
+    Adjustment
+    Transfer
+    Item
+    BarcodeItem
+  ).freeze
+
+  def initialize(organization, type)
+    @current_organization = organization
+    @type = type
+  end
+
+  def as_csv
+    return nil unless current_organization.present? && type.present?
+    return nil unless SUPPORTED_TYPES.include? type
```

Comment:
Maybe these should raise an error instead of returning `nil`

Reply:
I'm not sure putting them in the `app/models` is the correct place, since it's essentially a service object. 

What's the current best practice about that?